File: Week 6/cash.py
def main():
    while True:
        try:
            dollars = float(input("Change owed: "))
            if dollars >= 0:

                cents = round(dollars * 100)
                break
        except ValueError:
            pass

    coins = 0
    for value in [25, 10, 5, 1]:
        coins += cents // value
        cents %= value

    print(coins)

if __name__ == "__main__":
    main()

File: Week 1/cash/cash.c
#include <stdio.h>

int main(void)
{
    int cents;
    int coins = 0;

    do
    {
        printf("Change owed: ");
        if (scanf("%d", &cents) != 1)
        {
            while (getchar() != '\n');
            cents = -1;
        }
    }
    while (cents < 0);


    while (cents > 0)
    {
        if (cents >= 25)
        {
            cents -= 25;
            coins++;
        }
        else if (cents >= 10)
        {
            cents -= 10;
            coins++;
        }
        else if (cents >= 5)
        {
            cents -= 5;
            coins++;
        }
        else
        {
            cents -= 1;
            coins++;
        }
    }

    printf("%d\n", coins);

    return 0;
}
